x-pack/plugins/synthetics/public/apps/synthetics/components/step_details_page/step_waterfall_chart/waterfall/waterfall_marker/waterfall_markers.tsx

```diff
@@ -35,16 +34,36 @@ export function WaterfallChartMarkers() {
   );
 
   const annotations = useMemo(() => {
-    return Array.from(markerItemsByOffset.entries()).map(([offset, items]) => {
-      let uniqueIds = (items ?? [])
+    let lastOffset: number;
+    const recognizedMarkerItemsByOffset = Array.from(markerItemsByOffset.entries()).reduce(
+      (acc, [offset, items]) => {
+        // Remove unrecognized marks e.g. custom marks
+        const vitalMarkers = items.filter(({ id }) => getMarkersInfo(id, theme) !== undefined);
+
+        const hasMultipleMarksAtOffset = vitalMarkers.some(({ id }) => id !== LAYOUT_SHIFT);
+        const isLastOffsetTooClose = lastOffset && Math.abs(offset - lastOffset) < 100; // 100ms
+
+        // If offsets coincide or are too close, remove less important marks e.g. Layout Shift
+        const filteredItems =
+          hasMultipleMarksAtOffset || isLastOffsetTooClose
+            ? vitalMarkers.filter(({ id }) => id !== LAYOUT_SHIFT)
+            : vitalMarkers;
+
+        if (filteredItems.length) {
+          acc.set(offset, filteredItems);
+        }
+
+        lastOffset = offset;
+        return acc;
+      },
+      new Map<number, MarkerItems>()
+    );
```

Comment:
created a follow up issue to add a toggle to display custom user timings https://github.com/elastic/kibana/issues/148364